src/components/utils/test/thread_test.cc

```diff
@@ -257,6 +266,66 @@ TEST_F(ThreadTest, StartJoinStart_Success) {
   EXPECT_FALSE(thread_->is_running());
 }
 
+TEST_F(ThreadTest, Start_Success) {
+  // Default ThreadOptions means it is joinable
+  ASSERT_TRUE(thread_->start(threads::ThreadOptions()));
+
+  // Wait for the threadMain to signal that it has started
+  delegate_->WaitForThreadStart();
+  EXPECT_TRUE(delegate_->is_thread_started());
+  CheckThreadHasCompleted();
+}
+
+TEST_F(ThreadTest, StartTwice_Success) {
+  ASSERT_TRUE(thread_->start(threads::ThreadOptions()));
+
+  // Wait for the thread to notify it has started execution
+  delegate_->WaitForThreadStart();
+  EXPECT_TRUE(thread_->is_running());
+
+  // Try to start the same thread second time
+  EXPECT_TRUE(thread_->start());
+
+  // Check thread has not restarted
+  EXPECT_EQ(delegate_->is_thread_restarted(), 1);
+
+  // Wait for the thread to notify it is running
+  delegate_->WaitForThreadStart();
+  EXPECT_TRUE(thread_->is_running());
+
+  // Make the thread stop and wait for it
+  thread_->join(Thread::JoinOptionStop::kForceStop);
+  EXPECT_FALSE(thread_->is_running());
+}
+
+TEST_F(ThreadTest, StartStop_Success) {
+  ASSERT_TRUE(thread_->start(threads::ThreadOptions()));
+
+  // Wait for the thread to notify it has started execution
+  delegate_->WaitForThreadStart();
+  EXPECT_TRUE(thread_->is_running());
+
+  // Signal the thread to stop and wait for the execution
+  thread_->stop();
+  thread_->join(Thread::JoinOptionStop::kNoStop);
+  EXPECT_FALSE(thread_->is_running());
+}
+
+TEST_F(ThreadTest, StartWithCustomOptions_Success) {
+  bool is_joinable = true;
+
+  // Start thread with custom options (custom stack size & thread is joinable)
+  ASSERT_TRUE(
+      thread_->start(threads::ThreadOptions(kThreadStackSize, is_joinable)));
+
+  // Check thread was created with correct parameters
+  EXPECT_EQ(kThreadName, thread_->name());
+  EXPECT_EQ(kThreadStackSize, thread_->stack_size());
+  EXPECT_EQ(thread_->is_joinable(), is_joinable);
+
+  CheckThreadHasCompleted();
```

Comment:
@wolfylambova22 btw in CheckThreadHasCompleted() instead 
delegate_->WaitForThreadStart();
thread_->join(Thread::JoinOptionStop::kForceStop);
can't you use just one line:
thread_->join(Thread::JoinOptionStop::kNoStop);
That was the intention for introducing kNoStop.